tpm2/structures.go

```diff
@@ -1021,10 +1021,37 @@ type AuthCommand struct {
 	Auth       tpmutil.U16Bytes
 }
 
+// BDigest implements the structure of TPM2B_Digest as described in Part 2 Structures, Page 97
+type BDigest struct {
+	Size   uint16
+	Digest []byte
+}
+
 // TPMLDigest represents the TPML_Digest structure
 // It is used to convey a list of digest values.
 //This type is used in TPM2_PolicyOR() and in TPM2_PCR_Read()
 type TPMLDigest struct {
 	Count   uint32
-	Digests []tpmutil.U16Bytes
+	Digests []BDigest
+}
+
+// ToBytes converts the TPMLDigest structure into a byte slice
+func (list *TPMLDigest) ToBytes() ([]byte, error) {
```

Comment:
This looks like it is meant to be the Encode() method for serializing a TPMLDigest to the wire format. I think (but am not 100% sure) that if you implement a custom Encode() for a struct, tpmutil.Pack will "just work" and you won't have to touch PolicyOR as part of this PR.

Reply:
Note that if you implement a special Encode(), you probably also need to implement DecodeTPMLDigest.